components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,539 @@
+			return nil, err
+		}
+
+		err = m.c.sendCmd(setRampCmd)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return m, nil
+}
+
+// IsPowered returns if the motor is currently on or off.
+func (m *Motor) IsPowered(ctx context.Context, extra map[string]interface{}) (bool, error) {
+	return m.isOn, nil
+}
+
+// Close stops the motor and marks the axis inactive.
+func (m *Motor) Close() {
+	m.c.mu.Lock()
+	active := m.c.activeAxes[m.Channel]
+	m.c.mu.Unlock()
+	if !active {
+		return
+	}
+	err := m.Stop(context.Background(), nil)
+	if err != nil {
+		m.c.logger.Error(err)
+	}
+
+	m.c.mu.Lock()
+	defer m.c.mu.Unlock()
+	m.c.activeAxes[m.Channel] = false
+	for _, active = range m.c.activeAxes {
+		if active {
+			return
+		}
+	}
+	if m.c.port != nil {
+		err = m.c.port.Close()
+		if err != nil {
+			m.c.logger.Error(err)
+		}
+	}
+	globalMu.Lock()
+	defer globalMu.Unlock()
+	delete(controllers, m.c.serialDevice)
+}
+
+// Must be run inside a lock.
+func (m *Motor) configure(c *Config) error {
+	// Turn off the motor with opMixedDrive and a value of 64 (stop)
+	cmd, err := newCommand(m.c.address, singleForward, c.MotorChannel, 0x00)
+	if err != nil {
+		return err
+	}
+	err = m.c.sendCmd(cmd)
+	return err
+}
+
+// Must be run inside a lock.
+func (c *controller) sendCmd(cmd *command) error {
+	packet := cmd.ToPacket()
+	if c.testChan != nil {
+		c.testChan <- packet
+		return nil
+	}
+	_, err := c.port.Write(packet)
+	return err
+}
+
+// SetPower instructs the motor to go in a specific direction at a percentage
+// of power between -1 and 1. Scaled to MaxRPM.
+func (m *Motor) SetPower(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
+	if math.Abs(powerPct) < m.minPowerPct {
+		return m.Stop(ctx, extra)
+	}
+	if powerPct > 1 {
+		powerPct = 1
+	} else if powerPct < -1 {
+		powerPct = -1
+	}
+
+	m.opMgr.CancelRunning(ctx)
+	m.c.mu.Lock()
+	defer m.c.mu.Unlock()
+	m.isOn = true
+
+	rawSpeed := powerPct * maxSpeed
+	if math.Signbit(rawSpeed) {
+		rawSpeed *= -1
+	}
+
+	// Jog
+	var cmd commandCode
+	if powerPct < 0 {
+		// If dirFlip is set, we actually want to reverse the command
+		if m.dirFlip {
+			cmd = singleForward
+		} else {
+			cmd = singleBackwards
+		}
+	} else {
+		// If dirFlip is set, we actually want to reverse the command
+		if m.dirFlip {
+			cmd = singleBackwards
+		} else {
+			cmd = singleForward
+		}
+	}
+	c, err := newCommand(m.c.address, cmd, m.Channel, byte(int(rawSpeed)))
+	if err != nil {
+		return err
+	}
+	err = m.c.sendCmd(c)
+	return err
+}
+
+// GoFor moves an inputted number of revolutions at the given rpm, no encoder is present
+// for this so power is determined via a linear relationship with the maxRPM and the distance
+// traveled is a time based estimation based on desired RPM.
+func (m *Motor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
+	if m.maxRPM == 0 {
+		return errors.New("not supported, define max_rpm attribute != 0")
+	}
+
```

Comment:
please add this condition. it was added to all motors since you branched -- https://github.com/viamrobotics/rdk/commit/c780970085d142c794f7a23962c952078a706fe8

```
	if rpm == 0 {
		return motor.NewZeroRPMError()
	}
```